src/System.Private.CoreLib/shared/System/String.Manipulation.cs

```diff
@@ -1686,6 +1686,27 @@ public string Substring(int startIndex, int length)
             return InternalSubString(startIndex, length);
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public string Substring(Index startIndex)
+        {
+            int actualIndex = startIndex.IsFromEnd ? _stringLength - startIndex.Value : startIndex.Value;
```

Comment:
We are using `Length` property everywhere else in this file. Why do use the `_stringLength` field directly here?

Reply:
will fix that